panther_driver/src/panther_can.py

```diff
@@ -18,327 +18,337 @@ def __setattr__(self, name, value):
-                except canopen.SdoCommunicationError:
-                    rospy.loginfo(f'[{rospy.get_name()}] Failed to read TPDO configuration.')
-                    return False
-
-                motor_controller.can_node.tpdo[1].add_callback(
-                    lambda message, motor_controller=motor_controller: self._update_wheels_pos_cb(
-                        message, motor_controller
-                    )
-                )
-                motor_controller.can_node.tpdo[2].add_callback(
-                    lambda message, motor_controller=motor_controller: self._update_wheels_speed_cb(
-                        message, motor_controller
-                    )
-                )
-                motor_controller.can_node.tpdo[3].add_callback(
-                    lambda message, motor_controller=motor_controller: self._update_wheels_current_cb(
-                        message, motor_controller
-                    )
-                )
-                motor_controller.can_node.tpdo[4].add_callback(
-                    lambda message, motor_controller=motor_controller: self._update_wheels_flags_cb(
-                        message, motor_controller
-                    )
+        rospy.loginfo(f'[{rospy.get_name()}] Connected to the CAN bus.')
+
+    def can_connection_error(self) -> Generator[bool, None, None]:
+        for controller in self._motor_controllers:
+            if self._robot_driver_initialized:
+                yield controller.connection_error
+            else:
+                self._robot_driver_initialized = (
+                    rospy.Time.now() - self._can_init_time > rospy.Duration(10.0)
                 )
+                yield False
 
-            rospy.loginfo(f'[{rospy.get_name()}] TPDO configured successfuly.')
-            return True
+    def restart_roboteq_script(self) -> bool:
+        for controller in self._motor_controllers:
+            if not isinstance(controller, MotorControllerPDO):
+                raise TypeError("An SDO object does not require script reset.")
 
-    def _update_wheels_pos_cb(
-        self, message: canopen.pdo.Map, motor_controller: MotorController
-    ) -> None:
-        motor_controller.wheel_pos = [m.raw for m in reversed(message)]
+            if not controller.restart_script():
+                return False
 
-    def _update_wheels_speed_cb(
-        self, message: canopen.pdo.Map, motor_controller: MotorController
-    ) -> None:
-        motor_controller.wheel_vel = [m.raw for m in reversed(message)]
+        return True
 
-    def _update_wheels_current_cb(
-        self, message: canopen.pdo.Map, motor_controller: MotorController
-    ) -> None:
-        motor_controller.wheel_curr = [m.raw for m in reversed(message)]
+    def set_wheels_enc_velocity(self, velocity: List[float]) -> None:
+        vel = [velocity[:2], velocity[2:]]
+        for controller, v in zip(self._motor_controllers, vel):
+            controller.write_wheels_enc_velocity(v)
```

Comment:
```suggestion
    def set_wheels_enc_velocity(self, velocity: List[float]) -> None:
        for controller, vel in zip(self._motor_controllers, [velocity[:2], velocity[2:]]):
            controller.write_wheels_enc_velocity(vel)
```
Also, for some reason I dont like this method being called `set_wheels_enc_velocity` - `write` made more sense. This methods could be named the same for can interface and controller but I don't know how we feel about it